File: UIPeriodicTable/CPlusPlus/Resources/Components/code_uipromptwindow.cpp
// UIPromptWindow application source code
#include <Atomic/UI/UIEvents.h>
#include <Atomic/UI/UIPromptWindow.h>
#include "PeriodicApp.h"

void PeriodicApp::setup_uipromptwindow( UIWidget *layout, UIView *uiview )
{
    PODVector<UIWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUipromptwindowEvent ));
}

void PeriodicApp::HandlePromptCompleteEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace UIPromptComplete;

    AppLog( "UIPromptWindow event : the window " + eventData[P_TITLE].GetString()
            + " file was " + eventData[P_SELECTED].GetString()
            + ", the button pressed was " + eventData[P_REASON].GetString());
}

void PeriodicApp::HandleUipromptwindowEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    UIWidget* widget = static_cast<UIWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uipromptwindowcode" )
        {
            AppLog( "UIPromptWindow support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Components/code_uipromptwindow.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uipromptwindowlayout" )
        {
            AppLog( "UIPromptWindow support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_uipromptwindow.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() == "stringfinder" )
        {
            AppLog( "UIPromptWindow action : " + widget->GetId() + " was pressed ");
            UIWidget *someview = (UIWidget *)widget->GetView();
            UIPromptWindow *windowp = new  UIPromptWindow(context_, someview, widget->GetId());
            SubscribeToEvent( windowp, E_UIPROMPTCOMPLETE, ATOMIC_HANDLER(PeriodicApp, HandlePromptCompleteEvent ));
            windowp->Show( "WindowTitle", "Message in window", "preset value", 0, 0, 0);
        }
    }
}

File: UIPeriodicTable/CSharp/Resources/Components/code_uipromptwindow.cs
// UIPromptWindow application source code
using System;
using AtomicEngine;

public class code_uipromptwindow : CSComponent {

    public void Setup( UIWidget  layout )
    {
        var dest = new AtomicEngine.Vector<AtomicEngine.UIWidget>();
        layout.SearchWidgetClass( "TBButton", dest );
        for (var ii = 0; ii < dest.Size; ii++) {
            dest[ii].SubscribeToEvent<WidgetEvent> (dest [ii], HandleUipromptwindowEvent );
        }
    }

    private static void HandlePromptCompleteEvent ( UIPromptCompleteEvent ev )
    {
        AtomicMain.AppLog( "UIPromptWindow event : the window " + ev.Title
                           + " file was " + ev.Selected  + ", the button pressed was " + ev.Reason);
    }

    private static void HandleUipromptwindowEvent ( WidgetEvent ev )
    {
        UIWidget widget = (UIWidget)ev.Target;
        if ( widget.Equals(null)) return;
        if ( ev.Type == UI_EVENT_TYPE.UI_EVENT_TYPE_CLICK) {
            if (widget.GetId() == "uipromptwindowcode" ) {
                AtomicMain.AppLog( "UIPromptWindow support : " + widget.GetId() + " was pressed " );
                AtomicMain.ViewCode ( "Components/code_uipromptwindow.cs", widget.GetParent() );
            }
            if (widget.GetId() ==  "uipromptwindowlayout" ) {
                AtomicMain.AppLog( "UIPromptWindow support : " + widget.GetId() + " was pressed ");
                AtomicMain.ViewCode ( "Scenes/layout_uipromptwindow.ui.txt", widget.GetParent() );
            }

            if (widget.GetId() == "stringfinder" ) {
                AtomicMain.AppLog( "UIPromptWindow action : " + widget.GetId() + " was pressed ");
                UIWidget someview = (UIWidget  )widget.GetView();
                UIPromptWindow  windowp = new  UIPromptWindow( someview, widget.GetId());
                windowp.SubscribeToEvent<UIPromptCompleteEvent> (windowp, HandlePromptCompleteEvent );
                windowp.Show( "WindowTitle", "Message in window", "preset value", 0, 0, 0);
            }
        }
    }
}

